### PR TITLE
Add appID in SystemRequest to HMI

### PR DESCRIPTION
**Pull request by orhan-mehmedov**
Added hmi_appID as appID in SystemRequest to HMI.
Made appID mandatory=true in SystemRequest in HMI_API.xml.

Related to [APPLINK-30384](https://adc.luxoft.com/jira/browse/APPLINK-30384).

### CODE REVIEW

**Comment by orhan-mehmedov** [comment]
Please, review:
@vlantonov, @AGaliuzov, @Kozoriz , @anosach-luxoft, @dev-gh, @wolfylambova22
@LuxoftAKutsan, @dtrunov

File: src/components/interfaces/HMI_API.xml [hunk 1]
```diff
@@ -2356,8 +2356,8 @@
       <param name="fileName" type="String" maxlength="255" minlength="1" mandatory="true">
           <description>The path to file.</description>
       </param>
-      <param name="appID" type="Integer" mandatory="false">
-        <description>ID of application that requested this RPC.</description>
+      <param name="appID" type="Integer" mandatory="true">
```
Comment (dtrunov) [inline src/components/interfaces/HMI_API.xml]: @orhan-mehmedov Please change version according to https://adc.luxoft.com/confluence/display/APPLINK/API+xml+versioning+approach

File: src/components/application_manager/src/commands/mobile/system_request.cc [hunk 2]
```diff
@@ -578,9 +578,8 @@ void SystemRequest::Run() {
     msg_params[strings::file_name] = file_dst_path;
   }
 
-  if (mobile_apis::RequestType::PROPRIETARY != request_type) {
-    msg_params[strings::app_id] = (application->policy_app_id());
-  }
+  msg_params[strings::app_id] = application->hmi_app_id();
+
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @orhan-mehmedov You changed policy_app_id to hmi_app_id is it correct?

Comment (AGaliuzov) [inline src/components/application_manager/src/commands/mobile/system_request.cc]: @dtrunov have you read the HMI_API spec?. The policy appplication ID is a string whereas according to the api integer value is required

File: src/components/interfaces/HMI_API.xml (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/interfaces/HMI_API.xml]: @orhan-mehmedov is it rely should be mandatory within all policies?

Comment (orhan-mehmedov) [inline src/components/interfaces/HMI_API.xml]: @dtrunov fixed in [443f98e](https://github.com/smartdevicelink/sdl_core/pull/1082/commits/443f98ee2cbd27d3a33bb6d97907dd8243898f52)

**Comment by TMelnyk** [comment]
reviewed

**Pull request merged**